File: frontend/src/services/translation.ts
export async function translateText(text: string, targetLang: string) {
  const response = await fetch("http://localhost:8000/translate", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      text,
      target_lang: targetLang,
    }),
  });

  if (!response.ok) {
    throw new Error("Translation failed");
  }

  const data = await response.json();

  return data.translations[0].text;
}

File: Ml Model API/api/services/translation.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

DEEPL_API_KEY = os.getenv("DEEPL_API_KEY")

def translate_text(text: str, target_lang: str):
    url = "https://api-free.deepl.com/v2/translate"

    headers = {
        "Authorization": f"DeepL-Auth-Key {DEEPL_API_KEY}",
        "Content-Type": "application/json",
    }

    payload = {
        "text": [text],
        "target_lang": target_lang.upper(),
    }

    response = requests.post(
        url,
        headers=headers,
        json=payload
    )

    if response.status_code != 200:
        return {"error": response.text}

    return response.json()